### ISSUE TITLE
MPEG-TS subtitles

### ISSUE THREAD

**Issue by balderys** [mangui/flashls#147]
Hi @mangui,

is it possible to handle MPEG-TS subtitles (aka DVB/DVD subtitles) because I guess it is not supported by flash/OSMF itself? Note that I'm not talking about WebVTT subtitles added to HLS draft.

**Comment by jlacivita** [comment]
most production-level end-user facing live HLS streams w/ captions use the binary MPEG2 TS subtitles. pre-recorded / VOD HLS streams can use WebVTT, but for live, it's not realistic since all of the live encoders out there just spit out binary streams.

Ideally flashls would parse those packets and surface them, just like the video tag does.

**Comment by jlacivita** [comment]
Hmm, i might be mistaken...  I was under the impression this issue was asking the same thing as:
https://github.com/mangui/flashls/issues/11

which is 608/708 CCs.

**Comment by jlacivita** [comment]
@balderys: Turns out what i want is 608/708 CCs.  They are embedded in the MPEG2 TS fragments, though.  I'm not sure what you're asking about, but it could be the same thing or not? @mangui, do you know if they're the same?

If so, see my comment in #11, which suggests that you don't need to do the parsing of the 608/708 data since there are other plug-ins that do so.

**Comment by balderys** [comment]
@jlacivita 608CC are little bit different in comparison with standard full subtitles. I'm not sure how the 608CC are represented in TS but I guess it is the same format only in different PID.

**Comment by triwav** [comment]
The way I understand it, 608 captions are actually within the TS file itself whereas webvtt which seems to the alternative approach by Apple is a separate playlist referenced in the master playlist that is split up into chunks and is read in parallel. I think webvtt would be the easier route considering there aren't many open source tools that handle 608.

**Comment by jlacivita** [comment]
WebVTT is certainly easier, but most live encoders output 608 today.

I think flashls only needs to dispatch the onCaptionInfo events with the binary 608 info. There's no need for flashls to decode/render 608. That's something OSMF of jwPlayer should handle.

**Comment by kfeinUI** [comment]
Yes, this is a duplicate of #11 . 608/708 support for firing the onCaptionInfo events is the gap. Adobe packages a captioning lib for OSMF with its Media Server product. Sure it would make more sense to include such a valuable assistive technology with OSMF, but they instead are keeping it closed source and separate. Easy enough to work with at least as their not a lot to their public interface. I wrote a simple OSMF plugin wrapper with JavaScript bridge.

Anyway...as mentioned previously, this feature is quite valuable within the industry. Essential for live events since this data gets passed through the encoder and streamed. This does work with HLS on Apple's players. Not to be confused with the HLS support for WebVTT which may be streamed, but doesn't really have the encoder support.
